rai/node/node.cpp

```diff
@@ -3795,21 +3901,40 @@ bool rai::active_transactions::start (std::pair<std::shared_ptr<rai::block>, std
 bool rai::active_transactions::vote (std::shared_ptr<rai::vote> vote_a)
 {
 	std::shared_ptr<rai::election> election;
+	bool replay (false);
+	bool processed (false);
 	{
 		std::lock_guard<std::mutex> lock (mutex);
-		auto root (vote_a->block->root ());
-		auto existing (roots.find (root));
-		if (existing != roots.end ())
+		for (auto vote_block : vote_a->blocks)
 		{
-			election = existing->election;
+			rai::election_vote_result result { false, false };
```

Comment:
I think this initialization should happen in `election_vote_result` so we don't get the problem of uninitialized members if we use the class elsewhere down the line.